x-pack/test/functional/apps/dashboard/_async_dashboard.ts

```diff
@@ -0,0 +1,123 @@
+        const hitCount = parseInt(await PageObjects.discover.getHitCount(), 10);
+        expect(hitCount).to.be.greaterThan(0);
+      });
+    });
+
+    after(async () => {
+      await PageObjects.common.navigateToUrl('home', 'tutorial_directory/sampleData');
+      await PageObjects.home.removeSampleDataSet('flights');
+      const isInstalled = await PageObjects.home.isSampleDataSetInstalled('flights');
+      expect(isInstalled).to.be(false);
+      // await security.testUser.restoreDefaults();
+    });
+
+    it('should launch sample flights data set dashboard', async () => {
+      await appMenu.clickLink('Dashboard');
+      await PageObjects.dashboard.loadSavedDashboard('[Flights] Global Flight Dashboard');
+      await PageObjects.header.waitUntilLoadingHasFinished();
+      await PageObjects.timePicker.setCommonlyUsedTime('superDatePickerCommonlyUsed_Last_1 year');
+      await PageObjects.header.waitUntilLoadingHasFinished();
+
+      // check at least one visualization
+      await renderable.waitForRender();
+      log.debug('Checking pie charts rendered');
+      await pieChart.expectPieSliceCount(4);
+
+      await appMenu.clickLink('Discover');
+      await retry.try(async function() {
+        const hitCount = parseInt(await PageObjects.discover.getHitCount(), 10);
+        expect(hitCount).to.be.greaterThan(0);
+      });
+      await appMenu.clickLink('Dashboard');
+      await PageObjects.header.waitUntilLoadingHasFinished();
+      await renderable.waitForRender();
+      log.debug('Checking pie charts rendered');
+      await pieChart.expectPieSliceCount(4);
+    });
+
+    it('toggle from Discover to Dashboard attempt 1', async () => {
+      await appMenu.clickLink('Discover');
+      await retry.try(async function() {
+        const hitCount = parseInt(await PageObjects.discover.getHitCount(), 10);
+        expect(hitCount).to.be.greaterThan(0);
+      });
+      await appMenu.clickLink('Dashboard');
+      await PageObjects.header.waitUntilLoadingHasFinished();
+      await renderable.waitForRender();
+      log.debug('Checking pie charts rendered');
+      await pieChart.expectPieSliceCount(4);
+    });
+
+    it('toggle from Discover to Dashboard attempt 2', async () => {
+      await appMenu.clickLink('Discover');
+      await retry.try(async function() {
+        const hitCount = parseInt(await PageObjects.discover.getHitCount(), 10);
+        expect(hitCount).to.be.greaterThan(0);
+      });
+      await appMenu.clickLink('Dashboard');
+      await PageObjects.header.waitUntilLoadingHasFinished();
+      await renderable.waitForRender();
+      log.debug('Checking pie charts rendered');
+      await pieChart.expectPieSliceCount(4);
+    });
+
+    // since we're loading sameple data and not picking a time range which is guaranteed to
+    // contain all the sample data, we probably can't check all of these
+
+    // log.debug('Checking area, bar and heatmap charts rendered');
```

Comment:
@LeeDr, should we clean this up? Not sure if we should just remove it or try to uncomment?

Reply:
yes, let me take a shot at getting all the verifications passing in that test.